### ISSUE TITLE
improve coordinate handling in components

### ISSUE THREAD

**Issue by ngokevin** [aframevr/aframe-core#700]
position, rotation, scale, geometry.translate, look-at have coordinate properties used in slightly different ways. would be good to have a-frame make coordinate handling magic and be able to accept all '1 2 3' and 'x: 1, y: 2, z: 3' and `{x: 1, y: 2, z: 3}`.

**Comment by dmarcos** [comment]
That's a work for the schema
```schema: { myCoordinate { default: '0 0 0', type: v3 } }``` ?

**Comment by ngokevin** [comment]
Yeah schema, coerce, and component work. Might infer it if the default is an {x,y,z} object or if user specifies `type: coordinate`. I had a start on this but i need to rethink since it's a lot to change
